test/repository/searchResult.test.ts

```diff
@@ -0,0 +1,152 @@
+
+describe('list Search Result', () => {
+  describe('given only pagination params', () => {
+    it('returns all the data with pagination.', async () => {
+      const createdUser = await createUser(user);
+
+      const searchResultPayload = [
+        {
+          ...searchResult,
+          id: 'cb77e671-2447-4197-897d-540bc114df48',
+          userId: createdUser.id,
+          keyword: 'node',
+        },
+        {
+          ...searchResult,
+          id: 'a895bb89-b29d-47b1-96c9-3501b2f26d0c',
+          userId: createdUser.id,
+          keyword: 'nothing',
+        },
+        {
+          ...searchResult,
+          id: '37c5d10e-47e4-44c5-8d06-f25222377323',
+          userId: createdUser.id,
+          keyword: 'noted',
+        },
+        {
+          ...searchResult,
+          id: '743e9b2c-d335-40bc-bfe0-aef75a5092b7',
+          userId: createdUser.id,
+          keyword: 'nowhere',
+        },
+        {
+          ...searchResult,
+          id: 'fe6be24f-2507-4056-8597-175cdcb065d9',
+          userId: createdUser.id,
+          keyword: 'litter',
+        },
+      ];
+      await createSearchResult(searchResultPayload);
+      const paginationQuery = getPaginationParams({ maxRows: 3 });
+
+      const data = await listSearchResult({ paginationQuery });
+
+      expect(data.length).toEqual(3);
+    });
+  });
+
+  describe('given pagination and filter params', () => {
+    describe('given user have search results', () => {
+      it('returns the valid data with pagination after filter', async () => {
+        const createdUser = await createUser(user);
+
+        const searchResultPayload = [
+          {
+            ...searchResult,
+            id: 'cb77e671-2447-4197-897d-540bc114df48',
+            userId: createdUser.id,
+            keyword: 'node',
+          },
+          {
+            ...searchResult,
+            id: 'a895bb89-b29d-47b1-96c9-3501b2f26d0c',
+            userId: createdUser.id,
+            keyword: 'nothing',
+          },
+          {
+            ...searchResult,
+            id: '37c5d10e-47e4-44c5-8d06-f25222377323',
+            userId: createdUser.id,
+            keyword: 'noted',
+          },
+          {
+            ...searchResult,
+            id: '743e9b2c-d335-40bc-bfe0-aef75a5092b7',
+            userId: createdUser.id,
+            keyword: 'nowhere',
+          },
+          {
+            ...searchResult,
+            id: 'fe6be24f-2507-4056-8597-175cdcb065d9',
+            userId: createdUser.id,
+            keyword: 'litter',
+          },
+        ];
+        await createSearchResult(searchResultPayload);
+
+        const customQuery = getSearchResultQuery({ userId: createdUser.id, keyword: 'no' });
+        const paginationQuery = getPaginationParams({ maxRows: 3 });
+
+        const data = await listSearchResult({ customQuery, paginationQuery });
+
+        expect(data.length).toEqual(3);
+      });
+    });
+
+    describe('given user have no search Result', () => {
```

Comment:
```suggestion
    describe('given user have no search results', () => {
```

As you are already using the world "results" instead of "Result" [above](https://github.com/nimblehq/ic-nextjs-prabesh-dahal/pull/75/files#diff-ecc7ec255649de569928eaa447a954face611cedeaf5d9f51bd987b921e5b5deR59).